pkg/client/unversioned/clientcmd/merged_client_builder.go

```diff
@@ -112,8 +112,11 @@ func (config *DeferredLoadingClientConfig) ClientConfig() (*restclient.Config, e
 		//   "empty due to defaults"
 		// TODO: this shouldn't be a global - the client config rules should be
 		//   handling this.
-		defaultConfig, err := DefaultClientConfig.ClientConfig()
-		if err == nil && !reflect.DeepEqual(mergedConfig, defaultConfig) {
+		defaultConfig, defErr := DefaultClientConfig.ClientConfig()
+		if IsConfigurationInvalid(defErr) && !IsEmptyConfig(err) {
```

Comment:
So you still need to do the check against the default config in the non empty (err == nil) case.  IsConfigurationInvalid covers IsEmptyConfig already.